src/builder-utils.c

```diff
@@ -1836,3 +1836,50 @@ flatpak_version_check (int major,
 
   return FALSE;
 }
+
+gboolean
+appstream_version_check (int major,
+                         int minor,
+                         int micro)
+{
+  static int as_major = 0;
+  static int as_minor = 0;
+  static int as_micro = 0;
+
+  if (as_major == 0 &&
+      as_minor == 0 &&
+      as_micro == 0)
+    {
+      const char * argv[] = { "appstreamcli", "--version", NULL };
+      g_autoptr(GSubprocess) subp = NULL;
+      g_autofree char *out = NULL;
+
+      subp = g_subprocess_newv (argv, G_SUBPROCESS_FLAGS_STDOUT_PIPE, NULL);
+      g_subprocess_communicate_utf8 (subp, NULL, NULL, &out, NULL, NULL);
+
+      gchar **lines = g_strsplit (out, "\n", -1);
+      for (int i = 0; lines[i] != NULL; i++)
```

Comment:
```suggestion
      for (size_t i = 0; lines[i] != NULL; i++)
```
Nit picky, but indexing should really use size_t.